chainer/testing/backend.py

```diff
@@ -0,0 +1,126 @@
+import functools
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer.testing import attr
+
+
+class BackendConfig(object):
+
+    _props = [
+        ('use_cuda', False),
+        ('use_cudnn', 'never'),
+        ('cudnn_deterministic', False),
+        ('autotune', False),
+    ]
+
+    def __init__(self, params):
+        assert isinstance(params, dict)
```

Comment:
How about raising an exception instead since the constructor may be called by a user?

Reply:
Fixed. this class is not meant to be instantiated by a user, though.

Reply:
Thanks! Yes, I assumed that but it wasn't completely clear from the module path/class declaration.